src/transformation/visitors/return.ts

```diff
@@ -93,10 +91,10 @@ export function createReturnStatement(
 
     if (isInTryCatch(context)) {
         // Bubble up explicit return flag and check if we're inside a try/catch block
-        results.unshift(lua.createBooleanLiteral(true));
+        values.unshift(lua.createBooleanLiteral(true));
```

Comment:
Oh I see now why I was confused before, my bad. We should not be modifying input parameters, please revert.